lib/Date/Utility.pm

```diff
@@ -592,20 +590,17 @@ Returns the name of the current day. Example: Sunday
 
 =cut
 
+# 0..6: Sunday first.
+my @day_names = qw(Sunday Monday Tuesday Wednesday Thursday Friday Saturday);
+my %day_abbrev_to_num;
+foreach my $day_num (0 .. $#day_names) {
+    $day_abbrev_to_num{lc substr($day_names[$day_num], 0, 3)} = $day_num;
+}
```

Comment:
What you can do is this:

```perl
my %day_abbrev_to_num = map {
    my $day = $day_names[$_];
    (lc($day) => $_, lc(substr $day, 0, 3) =>$_, $_ => $_);
} 0..$#day_names;
```

That way you can pass full day names, abbreviated day names and day numbers (0..6) to the function below.